master/internal/command/command.go

```diff
@@ -352,13 +352,37 @@ func (c *command) readinessChecksPass(ctx *actor.Context, log sproto.ContainerLo
 	return len(c.readinessChecks) == 0
 }
 
+func (c *command) getState() State {
+	state := Pending
+	switch {
+	case c.container != nil:
+		switch c.container.State {
+		case container.Assigned:
+			state = Assigned
+		case container.Pulling:
+			state = Pulling
+		case container.Starting:
+			state = Starting
+		case container.Running:
+			state = Running
+		case container.Terminated:
+			state = Terminated
+		}
+	case c.exitStatus != nil:
+		state = Terminated
+	}
+	return state
+}
+
 func (c *command) toNotebook(ctx *actor.Context) (*notebookv1.Notebook, error) {
 	serviceAddress, err := generateServiceAddress(string(c.taskID))
 	if err != nil {
 		return nil, errors.Wrapf(err, "generating service address for %s", c.taskID)
 	}
+
 	return &notebookv1.Notebook{
 		Id:             ctx.Self().Address().Local(),
+		State:          c.getState().Proto(),
```

Comment:
Why are we not adding this too for other commands (regular commands, shells, etc..)?

Reply:
I will (in this pr). The rest are gonna be very similar and I wanted to confirm one set before applying it to the rest.

Reply:
although I now realize the changes were so small I should probably added it https://github.com/determined-ai/determined/pull/1958/commits/e0d4b20d7706eaa1a66d3a3806486b5ceebbe887